### DISCUSSION TITLE
How to set the AppScheduler misfire_grace_time parameter in stackstorm ?

### DISCUSSION THREAD

**Discussion by lemeur** [StackStorm/st2#5824] [Operations]
On one stackstorm installation I've got missing CronTimer events.

In the log I get these kind of entries:
`stackstorm st2timersengine[27662]: 2022-11-27 20:45:02,381 WARNING [-] Run time of job "St2Timer._emit_trigger_instance (trigger: cron[hour='20', minute='45'], next run at: 2022-11-28 20:45:00 UTC)" was missed by 0:00:02.381851`

After a little Googling, I think I need to set the misfire_grace_time parameter of the AppScheduler component used by Stackstorm.

Howerver I see no such option in the timersengine section of st2.conf, nor in the rule definition for the core.st2.CronTimer trigger.

Can someone point me to the correct way to set this parameter ?

**Comment by lemeur** [comment]
Here are the details of the technical environnement:
==> The server is installed on a Virtual Machine, which may explain why it is possible, under a high load on the hypervisor, that the VM misses "clock ticks".


StackStorm Version
st2 3.6.0, on Python 3.6.9
Distribution
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
RabbitM
rabbitmq-server 3.6.10-1ubuntu0.5
MongoDB
MongoDB shell version v4.0.28

**Comment by lemeur** [comment]
This is the same issue as https://github.com/StackStorm/st2/issues/5831

**Comment by lemeur** [comment]
For now, just scheduling the task at a different time has solved the issue. this is clearly related to some VM-Host load. 
However my sysadmin hasn't seen any CPU-Wait warning, there might be some I/O wait because the error occurred during the VM-backup window.

**Comment by jonathangull** [comment]
HI Team I am also facing this issue . Here are my environment details

st2 3.7.0, on Python 3.6.9
Distribution
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
MongoDB shell version v4.0.28
git version: af1a9dc12adcfa83cc19571cb3faba26eeddac92
OpenSSL version: OpenSSL 1.1.1  11 Sep 2018
allocator: tcmalloc
modules: none
build environment:
    distmod: ubuntu1804
    distarch: x86_64
    target_arch: x86_64
